security-oauth2/src/main/java/io/micronaut/security/oauth2/endpoint/authorization/response/DefaultAuthorizationErrorResponse.java

```diff
@@ -56,8 +60,33 @@ public DefaultAuthorizationErrorResponse(@Parameter HttpRequest<Map<String, Obje
     @NonNull
     @Override
     public ErrorCode getError() {
-        String name = responseData.get(JSON_KEY_ERROR).toUpperCase(Locale.ENGLISH);
-        return AuthorizationErrorCode.valueOf(name);
+        return getError(responseData);
+    }
+
+    @NonNull
+    static ErrorCode getError(ConvertibleMultiValues<String> responseData) {
+        String name = responseData.get(JSON_KEY_ERROR);
+        if (name != null) {
+            name = name.toUpperCase(Locale.ENGLISH);
+        }
+        try {
+            if (name != null) {
```

Comment:
Why not combining the 2 `name != null` branches?

Reply:
I have collapsed it as suggested https://github.com/micronaut-projects/micronaut-security/pull/1231/commits/59970647ac2ea3540c26cc3420f70a52a23fcde3